docs/api.md

```diff
@@ -2,202 +2,266 @@
-are listed here, and are definitely subject to change.  (Note that the
-default is `unknown`.)
-
-| **hardwareProfile** | **Root Device** |
-|---------------------|-----------------|
-| `unknown`           | /dev/sda        |
-|  `libvirt`          |  /dev/vda       |
-| `dell`              | HCTL: 0:0:0:0   |
-| `dell-raid`         | HCTL: 0:2:0:0   |
-
-*poweredOn* -- Boolean indicating whether the host is powered on. See
-*online*.
-
-*provisioning* -- Settings related to deploying an image to the host.
-
-*provisioning.state* -- The current state of any ongoing provisioning
-operation.  One of:
-  * *<empty string>* -- No provisioning is happening.
-  * *registration error* -- The details for the BMC for the host are
-    incorrect or incomplete and the host could not be managed.
-  * *registering* -- The details for the BMC for the host are being
-    checked.
-  * *match profile* -- The details for the hardware found on the host
-    are being compared against known profiles.
-  * *ready* -- The host is available to be consumed.
-  * *validation error* -- The details for the BMC for the host are
-    incorrect or incomplete and the host could not be managed.
-  * *provisioning* -- An image is being written to the host.
-  * *provisioning error* -- The image could not be written to the host.
-  * *provisioned* -- An image has been written to the host completely.
-  * *externally provisioned* -- The host has a consumer, but metal3
-    did not write an image to the host.
-  * *deprovisioning* -- The host storage is being wiped.
-  * *inspecting* -- The hardware details for the host are being
-    collected.
-  * *power management error* -- The host could not be powered on or
-    off, as requested through the *online* field setting.
-
-```
-apiVersion: metal3.io/v1alpha1
-kind: BareMetalHost
-metadata:
-  creationTimestamp: 2019-02-08T20:10:32Z
+  creationTimestamp: "2019-09-20T06:33:35Z"
   finalizers:
   - baremetalhost.metal3.io
-  generation: 9
-  name: example-baremetalhost
+  generation: 2
+  name: bmo-master-0
   namespace: bmo-project
-  resourceVersion: "1750818"
-  selfLink: /apis/metal3.io/v1alpha1/namespaces/bmo-project/baremetalhosts/example-baremetalhost
-  uid: 96837048-2bdd-11e9-8df7-525400f68198
+  resourceVersion: "22642"
+  selfLink: /apis/metal3.io/v1alpha1/namespaces/bmo-project/baremetalhosts/bmo-master-0
+  uid: 92b2f77a-db70-11e9-9db1-525400764849
 spec:
   bmc:
-    credentials:
-      name: bmc1-secret
-    address: ipmi://192.168.122.1:6233
+    address: ipmi://10.10.57.19
+    credentialsName: bmo-master-0-bmc-secret
+  bootMACAddress: 98:03:9b:61:80:48
+  consumerRef:
+    apiVersion: machine.openshift.io/v1beta1
+    kind: Machine
+    name: bmo-master-0
+    namespace: bmo-project
+  externallyProvisioned: true
+  hardwareProfile: default
+  image:
+    checksum: http://172.16.1.100/images/myOS.qcow2.md5sum
+    url: http://172.16.1.100/images/myOSv1/myOS.qcow2
```

Comment:
The paths in these values don't match. Should they?

Reply:
I guess they don't have to, but it was my intention that they'd match, fixing it, thanks!